fastly/erl.go

```diff
@@ -0,0 +1,342 @@
+//
+// API reference: https://developer.fastly.com/reference/api/vcl-services/rate-limiter/
+//
+// Supports these functions
+//   1) List rate limiters
+//   2) Create rate limiter
+//   3) Delete rate limiter
+//   4) Get rate limiter
+//   5) Update rate limiter
+//
+package fastly
+
+import (
+	"fmt"
+	"sort"
+	"time"
+)
+
+// Main definition of an Edge Rate Limiter
+type ERL struct {
+	Action             ERLActionEnum        `mapstructure:"action"`
+	ClientKey          []string             `mapstructure:"client_key"`
+	CreatedAt          *time.Time           `mapstructure:"created_at"`
+	DeletedAt          *time.Time           `mapstructure:"deleted_at"`
+	FeatureRevision    int                  `mapstructure:"feature_revision"` // 1..
+	HttpMethods        []ERLHttpMethodsEnum `mapstructure:"http_methods"`
+	ID                 string               `mapstructure:"id"`
+	LoggerType         ERLLoggerEnum        `mapstructure:"logger_type"`
+	Name               string               `mapstructure:"name"`
+	PenaltyBoxDuration int                  `mapstructure:"version"`  // 1..60
+	Response           *ERLResponseType     `mapstructure:"response"` // required if Action != Log
+	ResponseObjectName string               `mapstructure:"response_object_name"`
+	RpsLimit           int                  `mapstructure:"rps_limit"` // 10..10000
+	ServiceId          string               `mapstructure:"service_id"`
+	UpdatedAt          *time.Time           `mapstructure:"updated_at"`
+	UriDictionaryName  string               `mapstructure:"uri_dictionary_name"`
+	Version            int                  `mapstructure:"version"` // 1..
+	WindowSize         ERLWindowSizeEnum    `mapstructure:"window_size"`
+}
+
+// Type for ERL
+type ERLResponseType struct {
+	ERLStatus      int    `json:"status,omitempty"`
+	ERLContentType string `json:"content_type,omitempty"`
+	ERLContent     string `json:"content,omitempty"`
+}
+
+// Enum for ERL
+type ERLActionEnum string
+
+const (
+	ERLActionResponse       ERLActionEnum = "response"
+	ERLActionResponseObject ERLActionEnum = "response_object"
+	ERLActionLogOnly        ERLActionEnum = "log_only"
+)
+
+// Enum for ERL
+type ERLHttpMethodsEnum string
+
+const (
+	ERLGet     ERLHttpMethodsEnum = "GET"
+	ERLPut     ERLHttpMethodsEnum = "PUT"
+	ERLTrace   ERLHttpMethodsEnum = "TRACE"
+	ERLPost    ERLHttpMethodsEnum = "POST"
+	ERLHead    ERLHttpMethodsEnum = "HEAD"
+	ERLDelete  ERLHttpMethodsEnum = "DELETE"
+	ERLPatch   ERLHttpMethodsEnum = "PATCH"
+	ERLOptions ERLHttpMethodsEnum = "OPTIONS"
```

Comment:
Sorted and `Enum` suffix removed...
```suggestion
	ERLDelete  ERLHttpMethods = "DELETE"
	ERLGet     ERLHttpMethods = "GET"
	ERLHead    ERLHttpMethods = "HEAD"
	ERLOptions ERLHttpMethods = "OPTIONS"
	ERLPatch   ERLHttpMethods = "PATCH"
	ERLPost    ERLHttpMethods = "POST"
	ERLPut     ERLHttpMethods = "PUT"
	ERLTrace   ERLHttpMethods = "TRACE"
```